spec/puppet-server-lib/puppet/jvm/http_client_spec.rb

```diff
@@ -21,4 +21,19 @@
       expect(request_options.get_connect_timeout_milliseconds).to equal(connect_timeout)
     end
   end
+
+  context "requests" do
+    it "throw Ruby SocketErrors instead of Java exceptions" do
+      client = Puppet::Server::HttpClient.new("localhost", 0, {})
+      [lambda { client.get('/', nil) },
+       lambda { client.post('/', nil, nil) }
+      ].each do |request_block|
+        expect(&request_block).to raise_error { |error|
+          expect(error).to be_a(SocketError)
+          expect(error).to be_a(Puppet::Server::HttpClientError)
+          expect(error.message).to eq('com.puppetlabs.http.client.HttpClientException: Error executing http request')
+        }
```

Comment:
Pardon this test...but it violates The Wolfe Principle by spitting out `Connection refused` exception log messages.
I tried for 5 minutes to figure out a way to silence them, but it's an interop cluster****.

Suggestions welcome.